src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java

```diff
@@ -320,9 +357,11 @@ AND EXISTS (
      * @return all exercise-ids which belong to the course
      */
     @Query("""
-            SELECT e.id FROM Exercise e LEFT JOIN e.course c
+            SELECT e.id
+            FROM Exercise e
+                LEFT JOIN e.course c
             WHERE c.id = :courseId
-                """)
+            """)
     Set<Long> findAllIdsByCourseId(@Param("courseId") Long courseId);
```

Comment:
Same here?

Reply:
Not really, we don't return the Exercise itself. We could use a projection, but that would require using a Wrapper Class for the ID afaik. As far as I'm concerned this is the best way to implement this.